photon-api/src/main/scala/com/linkedin/photon/ml/function/glm/DistributedGLMLossFunction.scala

```diff
@@ -166,4 +166,11 @@ object DistributedGLMLossFunction {
       case _ => new DistributedGLMLossFunction(singleLossFunction, sparkContext, treeAggregateDepth)
     }
   }
+
+  def apply(
```

Comment:
Yes, I'll finish that cleanup by removing the creates. In general, we should use `apply` for factory methods as much as possible, as it is more concise, but there are limitations (scala 2.10 doesn't want 2 `apply` that define the same default values, for example).